### PR TITLE
[NFC][RemoveDIs] Remove conditional compilation for RemoveDIs

### PR DESCRIPTION
**Pull request by jmorse**
A colleague observes that switching the default value of LLVM_EXPERIMENTAL_DEBUGINFO_ITERATORS to "On" hasn't flipped the value in their CMakeCache.txt. This probably means that everyone with an existing build tree is going to not have support built in, meaning everyone in LLVM would need to clean+rebuild their worktree when we flip the switch on... which doesn't sound good.

So instead, just delete the flag and everything it does, making everyone build and run ~400 lit tests in RemoveDIs mode. None of the buildbots have had trouble with this, so it Should Be Fine (TM).

(Sending for review as this is changing various comments, and touches several different areas -- I don't want to get too punchy).

### CODE REVIEW

File: llvm/tools/llc/llc.cpp [hunk 1]
```diff
@@ -365,15 +365,10 @@ int main(int argc, char **argv) {
...
 
   // RemoveDIs debug-info transition: tests may request that we /try/ to use the
-  // new debug-info format, if it's built in.
-#ifdef EXPERIMENTAL_DEBUGINFO_ITERATORS
-  if (TryUseNewDbgInfoFormat) {
-    // If LLVM was built with support for this, turn the new debug-info format
-    // on.
+  // new debug-info format.
+  if (TryUseNewDbgInfoFormat)
+    // Turn the new debug-info format on.
```
Comment (SLTozer) [inline llvm/tools/llc/llc.cpp:370]: Nit, comments are counted when deciding whether an `if` needs braces; same applies everywhere else this is repeated.

**Review by SLTozer — COMMENTED** [review]
One (multiple?) nit from me, otherwise LGTM.

File: llvm/tools/llc/llc.cpp (hunk 1 above)
Comment (jmorse) [reply llvm/tools/llc/llc.cpp:370]: Fixed

**Comment by pogo59** [comment]
> hasn't flipped the value in their CMakeCache.txt.

Well, but that _never_ happens. You always have to delete CMakeCache.txt when you build with a parameter change. (IME CMakeCache.txt always takes precedence over what you say on the command line.)

**Pull request merged**